alerting/src/main/kotlin/org/opensearch/alerting/util/destinationmigration/DestinationMigrationUtilService.kt

```diff
@@ -64,17 +64,25 @@ class DestinationMigrationUtilService {
             try {
                 runningLock = true
 
-                val destinationsToMigrate = retrieveDestinationsToMigrate(client)
-                logger.info("Need to migrate ${destinationsToMigrate.size} destinations")
-                if (destinationsToMigrate.isEmpty()) {
+                val emailAccountsToMigrate = retrieveConfigsToMigrate(client, "email_account")
+                val emailGroupsToMigrate = retrieveConfigsToMigrate(client, "email_group")
+                val destinationsToMigrate = retrieveConfigsToMigrate(client, "destination")
+                val configsToMigrate = emailAccountsToMigrate + emailGroupsToMigrate + destinationsToMigrate
+                logger.info(
+                    "Need to migrate ${emailAccountsToMigrate.size} email accounts, " +
+                        "${emailGroupsToMigrate.size} email groups and " +
+                        "${destinationsToMigrate.size} destinations " +
+                        "(${configsToMigrate.size} configs total)"
+                )
+                if (configsToMigrate.isEmpty()) {
                     finishFlag = true
                     runningLock = false
                     return
                 }
-                val migratedDestinations = createNotificationChannelIfNotExists(client, destinationsToMigrate)
-                logger.info("Migrated ${migratedDestinations.size} destinations")
-                val failedDeletedDestinations = deleteOldDestinations(client, migratedDestinations)
-                logger.info("Failed to delete ${failedDeletedDestinations.size} destinations from migration process cleanup")
+                val migratedConfigs = createNotificationChannelIfNotExists(client, configsToMigrate)
```

Comment:
This might be scope of this PR, but what if the user who created the notification config, is deleted? Will it still migrate it? Or will it fail and keep retrying forever?

Reply:
You mean the user we're trying to inject in the thread context for the object is deleted?

Hmm, Security might handle that and treat it as no permissions? We'll want to test that when we test Alerting/Notif security integration. That's out of scope for this PR for now but a good callout.